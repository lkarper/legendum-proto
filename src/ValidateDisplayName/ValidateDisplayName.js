import React, { useEffect } from 'react';

const ValidateDisplayName = (props) => {

    const { displayNameError } = props;

    useEffect(() => {
        const { displayName, setDisplayNameError } = props;
        if (!displayName.trim().length) {
            setDisplayNameError('Displayname required.');
        } else {
            setDisplayNameError(null);
        }
    }, [props]);

    if (displayNameError) {
        return (
            <div role="alert">
                <p 
                    id="display-name-required"
                    style={{ color: 'darkred' }}
                >
                    {displayNameError}
                </p>
            </div>
        );
    }
    return <p id="display-name-required">Display name meets requirements.</p>;
}

export default ValidateDisplayName;