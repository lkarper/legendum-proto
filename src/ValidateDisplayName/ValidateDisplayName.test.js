import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateDisplayName from './ValidateDisplayName';

describe('ValidateDisplayName component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateDisplayName />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateDisplayName />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
