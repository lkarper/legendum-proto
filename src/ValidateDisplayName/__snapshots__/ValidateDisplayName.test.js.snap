// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateDisplayName component renders the UI as expected 1`] = `
<ValidateDisplayName
  displayName=""
  displayNameError=""
  setDisplayNameError={[Function]}
>
  <div
    role="alert"
  >
    <p
      id="display-name-required"
    >
      Display name meets requirements.
    </p>
  </div>
</ValidateDisplayName>
`;
